mirage/factories/collection-submission-action.ts

```diff
@@ -1,42 +1,276 @@
-    toState: faker.random.objectElement(Object.values(CollectionSubmissionReviewStates)),
-    actionTrigger: faker.random.objectElement(Object.values(CollectionSubmissionActionTrigger)),
+    creator: association() as CollectionSubmissionAction['creator'],
+    target: association() as CollectionSubmissionAction['target'],
+    afterCreate(collectionSubmissionAction: ModelInstance<MirageCollectionSubmissionAction>, server: Server) {
+        if (!collectionSubmissionAction.dateCreated) {
+            collectionSubmissionAction.update({dateCreated: getMinusDate(5)});
+        }
 
-    comment() {
-        return faker.lorem.sentence();
-    },
+        if (!collectionSubmissionAction.dateModified) {
+            collectionSubmissionAction.update({
+                dateModified: getAddedDate(
+                    3, new Date(collectionSubmissionAction.dateCreated),
+                ),
+            });
+        }
 
-    dateCreated() {
-        return faker.date.past(3);
-    },
+        if (collectionSubmissionAction.isAutogenerated) {
+            /**
+             * Accepted
+             */
+            if(collectionSubmissionAction.actionTrigger === CollectionSubmissionActionTrigger.Accept) {
+                if (!collectionSubmissionAction.comment) {
+                    collectionSubmissionAction.update({comment: 'I really love this project.'});
+                }
+                collectionSubmissionAction.update({fromState: CollectionSubmissionReviewStates.Pending});
 
-    dateModified() {
-        return faker.date.recent(5);
-    },
+                collectionSubmissionAction.update({toState: CollectionSubmissionReviewStates.Accepted});
 
-    creator: association() as CollectionSubmissionAction['creator'],
+                addPendingAction(
+                    server,
+                    collectionSubmissionAction,
+                    getMinusDate(
+                        10, new Date(collectionSubmissionAction.dateCreated),
+                    ),
+                );
+            /**
+             * Rejected
+             */
+            } else if(collectionSubmissionAction.actionTrigger === CollectionSubmissionActionTrigger.Reject) {
+                collectionSubmissionAction.update({fromState: CollectionSubmissionReviewStates.Pending});
+
+                collectionSubmissionAction.update({toState: CollectionSubmissionReviewStates.Rejected});
+
+                addPendingAction(
+                    server,
+                    collectionSubmissionAction,
+                    getMinusDate(
+                        10, new Date(collectionSubmissionAction.dateCreated),
+                    ),
+                );
+            /**
+             * Admin Removed
+             */
+            } else if(collectionSubmissionAction.actionTrigger === CollectionSubmissionActionTrigger.AdminRemove) {
+                collectionSubmissionAction.update({comment: 'I am making my project private'});
```

Comment:
I noticed in Accepted above you are only updating the comment if there is no comment, but here and below you're not making that check. Is that an intentional choice?